include/fiction/algorithms/physical_design/ortho_ordering_network.hpp

```diff
@@ -0,0 +1,722 @@
+                            ++latest_pos_inputs.y;
+                        }
+                    }
+                }
+
+#if (PROGRESS_BARS)
+                // update progress
+                bar(i);
+#endif
+            });
+
+        // Since the layout size is only known after placing all gates, the POs are placed after the main algorithm
+        place_outputs(layout, ctn, latest_pos, po_counter, node2pos);
+
+        // restore possibly set signal names
+        restore_names(ctn.color_ntk, layout, node2pos);
+
+        // statistical information
+        pst.x_size    = layout.x() + 1;
+        pst.y_size    = layout.y() + 1;
+        pst.num_gates = layout.num_gates();
+        pst.num_wires = layout.num_wires();
+
+        return layout;
+    }
+
+  private:
+    /**
+     * @param ntk Network with ordered PIs
+     * @param ps Design parameters for orthogonal P&R
+     * @param pst Design stats for orthogonal P&R
+     */
+
+    input_ordering_view<mockturtle::fanout_view<mockturtle::names_view<technology_network>>> ntk;
+
+    orthogonal_physical_design_params ps;
+    orthogonal_physical_design_stats& pst;  // NOLINT
+    // This line will suppress the warning cppcoreguidelines-avoid-const-or-ref-data-members
+
+    uint32_t po_counter{0};
+};
+
+}  // namespace detail
+
+/**
+ * A modification of the orthogonal algorithm, that addresses certain limitations of the original algorithm.
+ * The modifications aim to save area, reduce wiring and wire crossings, and optimize the placement of nodes.
+ * The first modification involves ordering Pis, which are connected as fan-ins to the same gate, and place them
+ * side-by-side. The reduction of the distance between such 'related' nodes reduces the wiring effort.
+ * The second drawback of the original algorithm relates to conflicts that arise when a PI has an outgoing edge
+ * colored south, as it would lead to forbidden routing over other Pis. The modified algorithm uses a combination
+ * of Pi ordering, conditional coloring, and a new placement rule for nodes colored 'south' to resolve conflicts
+ * while still adhering to coloring constraints. This enables the algorithm to utilize the rows of the Pis effectively,
+ * and minimize wire crossings.
+ *
+ * Furthermore, the algorithm addresses the challenges posed by inverters. The function 'inverter_substitution'
+ * substitutes two inverters at the fan-outs of a fan-out node with a single inverter at its fan-in.
+ * This substitution significantly reduces area, wiring, and wire crossings, as fewer gates need to be placed.
+ *
+ * May throw a high_degree_fanin_exception if `ntk` contains any node with a fan-in larger than 2.
+ *
+ * @tparam Lyt Desired gate-level layout type.
+ * @tparam Ntk Network type that acts as specification.
+ * @param ntk The network that is to place and route.
+ * @param ps Parameters.
+ * @param pst Statistics.
+ * @return A gate-level layout of type `Lyt` that implements `ntk` as an FCN circuit.
+ */
```

Comment:
```suggestion
/**
 * A modification of the `orthogonal` physical design algorithm that addresses certain limitations of the original.
 * The modifications aim to save area, reduce wiring and wire crossings, and optimize the placement of nodes.
 *
 * The first modification involves ordering PIs, which are connected as fan-ins to the same gate, and placing them
 * side-by-side. The reduction of the distance between such "related" nodes reduces the wiring effort and, thus, saves area.
 *
 * The second drawback of the original algorithm relates to conflicts that arise when a PI has an outgoing edge
 * colored "south", as it would lead to forbidden routing over other PIs. The modified algorithm uses a combination
 * of PI ordering, conditional coloring, and a new placement rule for nodes colored "south" to resolve conflicts
 * while still adhering to coloring constraints. This enables the algorithm to utilize PI rows in the layout more effectively, hence, minimizing wire crossings.
 *
 * Finally, this algorithm addresses the challenges posed by inverters. The function `inverter_substitution`
 * substitutes inverters at the outputs of each fan-out node with a single inverter at its input.
 * This substitution significantly reduces area, wiring, and wire crossings, because fewer gates need to be placed.
 *
 * May throw a `high_degree_fanin_exception` if `ntk` contains any node with a fan-in larger than `2`.
 *
 * @tparam Lyt Desired gate-level layout type.
 * @tparam Ntk Network type that acts as specification.
 * @param ntk The network that is to be placed and routed.
 * @param ps Parameters.
 * @param pst Statistics.
 * @return A gate-level layout of type `Lyt` that implements `ntk` as an FCN circuit.
 */
```

Reply:
I'm wondering how much we want to go into the implementation details here. Like the coloring of south and east. Can we make this more abstract?

Reply:
Also, can you add the reference to the paper here? See `orthogonal` or `exact` (or any other of our published algorithms) for an example.

Reply:
Furthermore, you can link your ISVLSI publication in `docs/publications.rst`.